drivers/adc/adc_stm32.c

```diff
@@ -811,6 +812,11 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_VREFINT) == channel_id) {
 		adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_VREFINT);
 	}
+
+	if (config->has_vbat_channel &&
```

Comment:
The reference manual suggests that the link to be enabled only when vbat is being sampled. Do you mind implementing that in this PR as well?

Reply:
correct, this part could be under
`#if defined(CONFIG_STM32_VBAT)`

Reply:
plus  LL_ADC_SetCommonPathInternalChAdd  to set LL_ADC_PATH_INTERNAL_VBAT